src/wasm.h

```diff
@@ -565,10 +568,35 @@ class Expression {
     assert(int(_id) == int(T::SpecificId));
     return (const T*)this;
   }
+
+  bool isConstExpression() const {
```

Comment:
We talked in person whether it would be better to use `Const` class for not only `[type].const` instructions but also for `ref.null` and `ref.func` instructions, now that they are constants too. That's a rather big change because `Const` is used everywhere, so we can think about it for a follow-up patch.  And I renamed `isConstExpression` to `isConstantExpression`, to make it a little less confusing from `Const` expressions.